code/obj/flock/structure/gnesisturret.dm

```diff
@@ -0,0 +1,163 @@
+
+	process()
+		if(!src.group)//if it dont exist it off
+			powered = FALSE
+			src.compute = 0
+			src.icon_state = "teleblocker-off"
+			return
+
+		if(src.flock.can_afford_compute(base_compute))
+			powered = TRUE
+			src.compute = -base_compute
+			src.icon_state = "teleblocker-on"
+		else//if there isnt enough juice
+			powered = FALSE
+			src.compute = 0
+			src.icon_state = "teleblocker-off"
+			return
+
+		//if we need to generate more juice, do so and up the compute cost appropriately
+		if(src.reagents.total_volume < src.reagents.maximum_volume)
+			if(src.flock.can_afford_compute(base_compute+fluid_gen_cost))
+				src.compute = -(base_compute + fluid_gen_cost)
+				src.reagents.add_reagent(fluid_gen_type, fluid_gen_amt)
+
+		if(src.reagents.total_volume >= fluid_shot_amt)
+			//shamelessly stolen from deployable_turret.dm
+			if(!src.target && !src.seek_target()) //attempt to set the target if no target
+				return
+			if(!src.target_valid(src.target)) //check valid target
+				src.target = null
+				return
+			else //GUN THEM DOWN
+				if(src.target)
+					SPAWN(0)
+						for(var/i in 1 to src.current_projectile.shot_number) //loop animation until finished
+							//flick("gnturret_fire",src)
+							muzzle_flash_any(src, 0, "muzzle_flash")
+							sleep(src.current_projectile.shot_delay)
+					shoot_projectile_ST_pixel_spread(src, current_projectile, target, 0, 0 , spread)
+
+	proc/seek_target()
+		var/list/target_list = list()
+		for (var/mob/living/C in range(src.range,src.loc))
+			if (!isnull(C) && src.target_valid(C))
+				target_list += C
+				var/distance = get_dist(C.loc,src.loc)
+				target_list[C] = distance
+
+		if (length(target_list)>0)
+			var/min_dist = 99999
+
+			for (var/mob/living/T in target_list)
+				if (target_list[T] < min_dist)
+					src.target = T
+					min_dist = target_list[T]
+
+			playsound(src.loc, "sound/misc/flockmind/flockdrone_door.ogg", 40, 1, pitch=0.5)
+
+		return src.target
+
+	proc/target_valid(var/mob/living/C)
+		var/distance = get_dist(get_turf(C),get_turf(src))
+
+		if(distance > src.range)
+			return FALSE
+		if (!C)
+			return FALSE
+		if(!isliving(C) || isintangible(C))
+			return FALSE
+		if (C.health < 0)
+			return FALSE
+		if (C.stat == 2)
+			return FALSE
+		if (!src.flock.isEnemy(C))
+			return FALSE
+		if (istype(C.loc,/obj/flock_structure/cage)) //already caged, stop shooting
+			return FALSE
+		if (istype(C,/mob/living/carbon/human))
+			var/mob/living/carbon/human/H = C
+			if (H.hasStatus(list("resting", "weakened", "stunned", "paralysis"))) // stops it from uselessly firing at people who are already suppressed. It's meant to be a suppression weapon!
+				return FALSE
+			if (H.reagents.has_reagent(fluid_gen_type,100)) //don't keep shooting at people who are already 1/3 flock
+				return FALSE
+		if (isflockmob(C))
+			return FALSE
+
+		return TRUE
+
+
+/datum/projectile/syringe/gnesis
```

Comment:
same as above, barb this

Reply:
https://github.com/stonepillars/goon-flock/pull/333